### ISSUE TITLE
Precision setting for currency field INR #,##,###.## 

### ISSUE THREAD

**Issue by sbkolate** [frappe/erpnext#3427]
We want to remove decimal because we deal in large number.
I have set #,### in INR, it will show ₹ 351,555
But it should be ₹ 3,51,555
Can you add  #,##,### option in Number Format?
So, We don't need to set Precision for every field.

**Comment by anandpdoshi** [comment]
You can change the precision in the currency record to 0 and see if it works out.

**Comment by sbkolate** [comment]
@anandpdoshi for currency and float we can set only 1-9 precision.
0 is not option. 

![currency](https://cloud.githubusercontent.com/assets/6648915/8024451/05d9e5e0-0d52-11e5-8deb-0d072998e6a7.png)

**Comment by anandpdoshi** [comment]
Go to List of Currencies and change the precision to 0 for INR. This will changer he precook everywhere. But it will affect all the rounding in calculations. 


Or we will have to add the option for 0 precision. CAn you send a pull request. 


Thanks,Anand Doshi.

**Comment by sbkolate** [comment]
There is no precision in Currency List.

Again I will explain my issue.
There is number format in currency list.
If I set  number format  #,###, it will remove precision. but since we are dealing in large number we want number format  #,##,### to show ₹ 3,25,000 instead of ₹ 325,000
This is standard for INR.

There is two pull request.
1) For number format  #,##,###
2) For adding 0 precision option for float and currency. 

Let me know is this right, I will send pull request for this.

**Comment by rmehta** [comment]
Yeah send pull-request. In general, its better to discuss over pull-request than Issue

**Comment by Tropicalrambler** [comment]
for Purchase Receipt or Purchase Invoice,   when entering amount and rate, the calculations according to price do not exactly match up.  Temporarily I use two solutions:  

1. Apply a discount to the line item to force a match in the QTY x PRICE (Amount x rate). Sometimes I use a negative discount.  Sometimes I use a positive discount. This process is by trial and error.
2. If method #1 does not work, I add a tax line item, with actual amount, added or deducted, depending on the case, (usually add).  I enter the differing amount to make the total Purchase Receipt or Purchase Invoice match the correct, printed amount.

I was observing that the field in ITEM-PRICE  only handles two decimal points, which I believe has a huge impact in the calculations per DocType.

May I suggest increasing the precision in the calculation (or display)?  Is this done in the javascript field?  I assume one would have to do float() for the numeric values involved in price.  

I wonder if this is done in the .json file or in the .py for the doctype **item_price** erpnext/erpnext/stock/doctype/item_price/

Thanks

**Comment by mafrica** [comment]
I am working in Tanzanian shillings (about 2000 per 1 USD).  At the moment I am testing the payroll functionality and despite setting the system wide format for currency etc to #,### I find the salary slip calculations insist on working to and printing to 2 decimal places of shillings - a vanishingly small amount which can confused our employees.

I have set precision to blank as the other available options of 2+ all seem inappropriate.

Is this a bug or a failing in my understanding?
